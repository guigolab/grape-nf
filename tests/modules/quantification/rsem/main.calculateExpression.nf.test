nextflow_process {

    name "Test Process quantify"
    script "modules/quantification/rsem/main.nf"
    process "calculateExpression"

    tag "module"

    setup {
        run("index") {
            script "modules/quantification/rsem/main.nf"
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa.gz")
                input[1] = file("${baseDir}/data/annotation.gtf.gz")
                """
            }
        }
    }

    test("Should quantify with RSEM") {

        when {
            process {
                """
                input[0] = index.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE", false ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should quantify with RSEM converting the input BAM file") {

        when {
            process {
                """
                input[0] = index.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE", true ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    // >>> TEST COMMENTED BECAUSE RSEM 1.3.3 FAILS WITH --calc-ci
    // >>> see https://github.com/deweylab/RSEM/issues/134
    // >>> DO NOT CLEAN SNAPSHOT!
    // test("Should quantify with RSEM (with credibility intervals)") {

    //     when {
    //         params {
    //             rsemCalcCI = true
    //         }
    //         process {
    //             """
    //             input[0] = index.out
    //             input[1] = Channel.from([
    //                 [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE", false ]
    //             ])
    //             """
    //         }
    //     }

    //     then {
    //         assert process.success
    //         assert snapshot(process.out).match()
    //     }

    // }

}
