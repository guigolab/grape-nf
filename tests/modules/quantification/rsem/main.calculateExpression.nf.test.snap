{
    "Should quantify with RSEM converting the input BAM file": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T13:07:20.24352"
    },
    "Should quantify with RSEM": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T16:57:55.746438"
    }
}